synapse_invite_checker/config.py

```diff
@@ -27,3 +27,4 @@ class InviteCheckerConfig:
     federation_localization_url: str = ""
     gematik_ca_baseurl: str = ""
     tim_type: TimType = TimType.PRO
+    allowed_room_versions: list = field(default_factory=list)
```

Comment:
How is this different from this?

```suggestion
    allowed_room_versions: list[str] = []
```

Reply:
From what I understand, setting a class variable with a mutable object(which is what `[]` is in this case, I believe it's called a literal?) contaminates any use of `[]` later. Same reason we can't use `[]` as a default for kwargs, it is bad. So for dataclasses, we use the `field()` with `default_factory=list` to cause the generated dunder method of `__init__()` generate a fresh empty `list` for us! :grin: 

(This one burned me a few years ago, hence my being conscious of it now)

Reply:
Makes sense.